core/integration/module_test.go

```diff
@@ -3395,6 +3395,70 @@ func (t *Toplevel) SayHello(ctx context.Context) (string, error) {
 	require.JSONEq(t, `{"toplevel":{"sayHello":"hello!"}}`, out)
 }
 
+func TestModuleGoCustomScalar(t *testing.T) {
+	t.Parallel()
+
+	var logs safeBuffer
+	c, ctx := connect(t, dagger.WithLogOutput(io.MultiWriter(os.Stdout, &logs)))
+
+	ctr := c.Container().From(golangImage).
+		WithMountedFile(testCLIBinPath, daggerCliFile(t, c))
+
+	ctr = ctr.
+		WithWorkdir("/toplevel/foo").
+		With(daggerExec("init", "--name=foo", "--sdk=go", "--source=.")).
+		WithNewFile("main.go", dagger.ContainerWithNewFileOpts{
+			Contents: `package main
+			
+import "fmt"
+
+type Foo struct {}
+
+type CustomString string
+type CustomInt int
+type CustomBool bool
```

Comment:
You're allowing users to set their own custom scalars, do you see any issues with that @sipsma and @vito?

Reply:
As discussed in discord, I've removed the custom scalars part of this patch - I'm not super fussed on this, so just going to remove it for now. If there's demand, I can always bring it back in https://github.com/dagger/dagger/compare/main...jedevc:dagger:add-custom-scalar-kind.